governance/program/src/instruction.rs

```diff
@@ -164,25 +154,36 @@ pub enum GovernanceInstruction {
     ///   1. `[writable]` Proposal account. PDA seeds ['governance',governance, governing_token_mint, proposal_index]
     ///   2. `[writable]` Governance account
     ///   3. `[writable]` TokenOwnerRecord account of the Proposal owner
-    ///   4. `[signer]` Governance Authority (Token Owner or Governance Delegate)
-    ///   5. `[signer]` Payer
-    ///   6. `[]` System program
-    ///   7. `[]` Rent sysvar
-    ///   8. `[]` Clock sysvar
-    ///   9. `[]` Optional Realm Config
-    ///   10. `[]` Optional Voter Weight Record
+    ///   4. `[]` Governing Token Mint the Proposal is created for
+    ///   5. `[signer]` Governance Authority (Token Owner or Governance Delegate)
+    ///   6. `[signer]` Payer
+    ///   7. `[]` System program
+    ///   8. `[]` Rent sysvar
+    ///   9. `[]` Clock sysvar
+    ///   10. `[]` Optional Realm Config
+    ///   11. `[]` Optional Voter Weight Record
     CreateProposal {
         #[allow(dead_code)]
         /// UTF-8 encoded name of the proposal
         name: String,
 
         #[allow(dead_code)]
-        /// Link to gist explaining proposal
+        /// Link to a gist explaining the proposal
         description_link: String,
 
         #[allow(dead_code)]
-        /// Governing Token Mint the Proposal is created for
-        governing_token_mint: Pubkey,
+        /// Proposal vote type
+        vote_type: VoteType,
+
+        #[allow(dead_code)]
+        /// Proposal options
+        options: Vec<String>,
+
+        #[allow(dead_code)]
+        /// Indicates whether the proposal has the deny option
+        /// A proposal without the rejecting option is a non binding survey
```

Comment:
I introduced the explicit `Deny` option and the distinction between `executable `/`non executable` proposals as a security guard. Without that it would be possible to create a malicious proposal with only `Yes option(s)` and executable instructions which could clear the `yes_vote_threshold` but the majority of the community could be still against it and it wouldn't be able to reject the proposal if  only positive choices were available. 

The flag makes also possible to create `non executable` proposal with `reject all` option. Note this is not the same as `non of the above` choice in multi option vote because of the direction of the vote which goes against the other choices.

Given the fact instructions are stored on different accounts it also makes surveys cheaper to create because we save on the rent.

Reply:
Ah I see, that explains it, thanks!